core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -643,6 +643,229 @@ class WorkspaceService(
       }
     }
 
+  def updateWorkspaceBillingProject(workspaceName: WorkspaceName, newBillingProjectName: String): Future[Workspace] = {
+    val sourceBillingProjectName = RawlsBillingProjectName(workspaceName.namespace)
+    val destBillingProjectName = RawlsBillingProjectName(newBillingProjectName)
+    for {
+      _ <-
+        if (workspaceName.namespace == newBillingProjectName)
+          Future.failed(
+            RawlsExceptionWithErrorReport(
+              ErrorReport(StatusCodes.BadRequest, s"Workspace billing is already set to $newBillingProjectName")
+            )
+          )
+        else Future.unit
+      sourceBillingProject <- getBillingProjectContext(sourceBillingProjectName)
```

Comment:
It looks like this only gets GCP billing projects, so `requireBillingProjectIsGCP` does not get reached if either the source or destination billing is in Azure.